docs/distribution/README.md

```diff
@@ -0,0 +1,923 @@
+
+**Pros with this approach:**
+
+* The signature artifact upload and role check are coupled to a signature API
+
+**Cons with this approach:**
+
+* Signature upload, which is just another artifact type, is uploaded differently than other artifacts.
+
+### Linking Signatures - Option 4: Utilize OCI Index PUT
+
+Utilizing the OCI Index, the manifest list is used to track dependencies.
+
+**Pros with this approach:**
+
+* Existing Index parsing logic is used to track despondencies
+* No additional distribution-spec APIs are required
+
+**Cons with this approach:**
+
+* OCI Index would need to support a config object:  [Add Index Support for Artifact Type #806](https://github.com/opencontainers/image-spec/issues/)
+
+## Signature Discovery
+
+Once a signature artifact is in a registry and linked to its target artifact, how is it retrieved?
+
+The following options are offered:
+
+1. [Option 1: Signature Listing API](#signature-discovery---option-1-rest-api-standard-paging)
+1. [Option 2: Generic Reference Listing API](#signature-discovery---option-2-generic-reference-listing-api)
+
+### Signature Discovery - Option 1: Signature Listing API
+
+Similar to the [_tags api][tags-api], a new signatures API is proposed. The signatures API uses the digest as the path to find all signatures related to said digest.
+In the below example, the `net-monitor:v1` tag has a digest of: `sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042`
+
+```HTTP
+GET /v2/<name>/manifests/sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042/signatures/
+```
+
+The response will be in the following format:
+
+```HTTP
+200 OK
+Content-Type: application/json
+{
+  "digest": "sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+  "signatures": [
+    {
+      "digest": "sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+      "mediaType": "application/vnd.oci.image.index.v1+json",
+      "size": "1024",
+      "signature-typ": "x509"
+    },
+    {
+      "digest": "sha256:007170c33ebc4a74a0a554c86ac2b28ddf3454a5ad9cf90ea8cea9f9e75a153b",
+      "mediaType": "application/vnd.oci.image.index.v1+json",
+      "size": "1025",
+      "signature-typ": "x509"
+    }
+  ]
+}
+```
+
+### Signature Discovery - Option 2: Generic Reference Listing API
+
+A slight alternative to the signatures API is to provide a generic reference listing API, where a paged collection of references are returned.
+
+In the below example, the `net-monitor:v1` tag has a digest of: `sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042`
+
+```HTTP
+GET /v2/<name>/manifests/sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042/references/
+```
+
+The response will be in the following format:
+
+```HTTP
+200 OK
+Content-Type: application/json
+{
+  "digest": "sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+  "references": [
+    {
```

Comment:
If purpose of the reference API is generic, I don't see why `config-mediaType` is needed in the response.

If `config-mediaType` is removed this generic API is not very helpful for signature discovery.

Reply:
Yup, agreed. This is definitely a work in progress. 
If folks continue to agree with storing signatures as index.config objects, and add config to index, then this is the next place we'll be focusing: what should this api look like. And, can it be generic? If it's generic, and only provides the artifact type, how do we know which of the 4 signatures we actually want? Or, do we need to pull all 4 (n) and decide after they're pulled. 
me: 👎